.gitmodules

```diff
@@ -25,3 +25,6 @@
 [submodule "third_party/benchmark"]
 	path = third_party/benchmark
 	url = https://github.com/google/benchmark
+[submodule "third_party/parallel-hashmap"]
+	path = third_party/parallel-hashmap
+	url = https://github.com/greg7mdp/parallel-hashmap.git
```

Comment:
@danpat so this is an Apache 2.0 license. we've got an MIT license. what do we need to do to make sure we are following the rules of the two licenses?

Reply:
Did we choose this because its easier to integrate (header only)?

There are other hashmaps which are equally performant for integral keys (the current usecase) as well as for other key types like strings. I found this blog on comparing hashmap performance very thorough:
https://martin.ankerl.com/2019/04/01/hashmap-benchmarks-01-overview/

Seems like `absl::flat_hash_map` can do the job as well, if integrating is not too much of a pain.

Reply:
@mandeepsandhu re: header-only: yes, that's my understanding. @danpat can clarify.

Pulling in absl was fairly straightforward CMake. See https://github.com/valhalla/valhalla/compare/master...mookerji:mookerji-hashmaps.

Reply:
The original choice was because it was a trivial `git submodule add`, and I was trying things out in between changing daipers and helping with Lego.

Now that I've had a bit more time to understand the code hotspots, I've pulled the third-party libraries out completely - the performance gains can be had with some fairly small changes just within our code.

It also avoids having to think about the implications of adding an Apache 2.0 submodule to the project (which both `absl` and `parallel-hashmap` are).  I hadn't checked that ahead of time.